server/routes/visitors.ts

```diff
@@ -27,40 +27,40 @@ export default function routes(router: Router, prisonerVisitorsService: Prisoner
 
   router.post(
     '/:offenderNo',
-    body('visitors')
-      .notEmpty()
-      .withMessage('No visitors selected')
-      .bail()
-      .custom((value: string, { req }) => {
-        const selected = [].concat(value)
-
-        req.session.visitorList = req.session.visitorList.map((visitor: VisitorListItem) => {
-          const newVisitor = visitor
-          newVisitor.selected = selected.includes(visitor.personId.toString())
-
-          return newVisitor
-        })
-
-        if (selected.length > 3) {
-          throw new Error('Select no more than 3 visitors with a maximum of 2 adults')
-        }
-
-        const adults = req.session.visitorList
-          .filter((visitor: VisitorListItem) => selected.includes(visitor.personId.toString()))
-          .reduce((count: number, visitor: VisitorListItem) => {
-            return visitor.adult ? count + 1 : count
-          }, 0)
-
-        if (adults === 0) {
-          throw new Error('Add an adult to the visit')
-        }
-
-        if (adults > 2) {
-          throw new Error('Select no more than 2 adults')
-        }
-
-        return selected
-      }),
+    body('visitors').custom((value: string, { req }) => {
+      const selected = [].concat(value)
+
+      req.session.visitorList = req.session.visitorList.map((visitor: VisitorListItem) => {
+        const newVisitor = visitor
+        newVisitor.selected = selected.includes(visitor.personId.toString())
+
+        return newVisitor
+      })
+
+      if (value === undefined) {
+        throw new Error('No visitors selected')
+      }
+
+      if (selected.length > 3) {
+        throw new Error('Select no more than 3 visitors with a maximum of 2 adults')
+      }
+
+      const adults = req.session.visitorList
+        .filter((visitor: VisitorListItem) => selected.includes(visitor.personId.toString()))
+        .reduce((count: number, visitor: VisitorListItem) => {
+          return visitor.adult ? count + 1 : count
```

Comment:
Could potentially change to:
`return visitor.adult ?? true ? count + 1 : count`
which I think would stop visitors with no date of birth (so `adult` property is `undefined`) causing issues by effectively counting them as adults - but probably need confirmation on what it *should* do

Reply:
👍